### PR TITLE
(PDK-1354) Default template ref for custom templates should always be master

### PR DESCRIPTION
**Pull request by rodjek**

### CODE REVIEW

File: lib/pdk/util/template_uri.rb [hunk 1]
```diff
@@ -207,14 +207,21 @@ def self.templates(opts)
...
+
         if PDK::Util.development_mode?
           'master'
         else
-          PDK::TEMPLATE_REF
+          uri.default? ? PDK::TEMPLATE_REF : 'master'
         end
       end
 
+      def default_template_ref
```
Comment (glennsarti) [inline lib/pdk/util/template_uri.rb]: Hrmmm well..this is a little confusing.  class vs instance name with different method signatures :-(

Comment (rodjek) [reply lib/pdk/util/template_uri.rb]: Yeah, we don't really need the instance method, so i've removed it

File: lib/pdk/util/template_uri.rb [hunk 2]
```diff
@@ -207,11 +207,14 @@ def self.templates(opts)
       end
 
       # @returns String
-      def self.default_template_ref
+      def self.default_template_ref(uri = nil)
```
Comment (glennsarti) [inline lib/pdk/util/template_uri.rb]: This seems a little complex. Maybe this is better (not sure)?

``` ruby
def self.default_template_ref(uri = nil)
  return 'master' if PDK::Util.development_mode?
  return PDK::TEMPLATE_REF if uri.nil?  # The default url always uses PDK::TEMPLATE_REF as the ref
  uri = new(uri) unless uri.is_a?(self)
  uri.default? ? PDK::TEMPLATE_REF : 'master'
end

File: spec/unit/pdk/util/template_uri_spec.rb [hunk 3]
```diff
@@ -161,7 +161,7 @@
 
           it 'returns the specified template' do
             allow(Gem).to receive(:win_platform?).and_return(true)
-            expect(template_uri.to_s).to eq("C:\\cli-templates##{described_class.default_template_ref}")
+            expect(template_uri.to_s).to eq('C:\cli-templates#master')
```
Comment (glennsarti) [inline spec/unit/pdk/util/template_uri_spec.rb]: nit: Although it's not required, I've found just always using double slash makes it far less confusing e.g. `\'` and `\\` is still are valid escape sequences in single quoted strings.

File: spec/unit/pdk/util/template_uri_spec.rb [hunk 4]
```diff
@@ -302,37 +302,45 @@
   end
 
   describe '.default_template_ref' do
-    subject { described_class.default_template_ref }
+    subject { described_class.default_template_ref(uri) }
```
Comment (glennsarti) [inline spec/unit/pdk/util/template_uri_spec.rb]: I don't see any tests for `described_class.default_template_ref()` anywhere.  Should these be added?

File: lib/pdk/util/template_uri.rb (hunk 2 above)
Comment (scotje) [reply lib/pdk/util/template_uri.rb]: Yeah I think @glennsarti 's guard clause version is clearer.

File: spec/unit/pdk/util/template_uri_spec.rb (hunk 4 above)
Comment (rodjek) [reply spec/unit/pdk/util/template_uri_spec.rb]: I can add an explicit `nil` test if you want, but it's just going to be duplicating the tests here https://github.com/puppetlabs/pdk/blob/e8797f535cb55b2fc29b195f80e54db7f5bcf8eb/spec/unit/pdk/util/template_uri_spec.rb#L331-L349

Comment (glennsarti) [reply spec/unit/pdk/util/template_uri_spec.rb]: The only thing I would be looking for is:
`expect(described_class.default_template_ref()).to eq(PDK::TEMPLATE_REF)`

**Pull request merged**